components/SignUpComponent.js

```diff
@@ -0,0 +1,69 @@
+import HomePage from "../pageobjects/BasePage";
+import { expect } from "@playwright/test";
+
+export default class SignUpComponent extends HomePage {
+  constructor(page) {
+    super(page);
+    this.button = this.page.locator(".hero-descriptor button");
+    this.modal = this.page.locator('[role="document"]');
+    this.name = this.modal.locator("#signupName");
+    this.lastName = this.modal.locator("#signupLastName");
+    this.email = this.modal.locator("#signupEmail");
+    this.password = this.modal.locator("#signupPassword");
+    this.repeatPassword = this.modal.locator("#signupRepeatPassword");
+    this.submit = this.modal.getByRole("button", {
+      name: "Register",
+    });
+    this.errorMessage = this.modal.locator(".invalid-feedback p");
+  }
+
+  async fillRegistrationForm(userName, userLastName, userEmail, password) {
+    await this.name.fill(userName);
+    await this.lastName.fill(userLastName);
+    await this.email.fill(`aqa+${userEmail}`);
+    await this.password.fill(password);
+    await this.repeatPassword.fill(password);
+    await this.submit.click();
+  }
+
+  async checkRegistrationFormWithInvalidUserName() {
+    const userName = " ";
+    await this.name.fill(userName);
+    this.name.focus();
+    this.name.blur();
+  }
+
+  async checkRegistrationFormWithInvalidUserLastName() {
+    const userLastName = " ";
+    await this.lastName.fill(userLastName);
+    this.lastName.focus();
+    this.lastName.blur();
+  }
+
+  async checkRegistrationFormWithInvalidUserEmail() {
+    const userEmail = "somename.com";
+    await this.email.fill(userEmail);
+    this.email.focus();
+    this.email.blur();
+  }
+
+  async checkRegistrationFormWithInvalidPassword() {
+    const password = " ";
+    await this.password.fill(password);
+    this.password.focus();
+    this.password.blur();
+  }
+
+  async checkRegistrationFormWithInvalidRepeatPassword() {
+    await this.password.fill("123qwe123QWE");
+    await this.repeatPassword.fill("123qwe123");
+    this.repeatPassword.focus();
+    this.repeatPassword.blur();
+  }
+
+  async checkErrorMessage(number, errorText) {
```

Comment:
It would be better to find error text for particular import, this implementation is not reliable.

Imagine situation

FE devs have broken form validation. You have entered invalid values and for one input error is not shown. Your method will find next error message related to other input. 
Your test will fail. But with strange error.

And even worse scenario. Validation is completely broken and the error text you are referring to has same value as you expect but in fact it is shown for not related input.
As result there is a bug and your test is passed